src/lib/datepicker/datepicker-input.ts

```diff
@@ -329,12 +329,19 @@ export class MatDatepickerInput<D> implements AfterContentInit, ControlValueAcce
     this._lastValueValid = !date || this._dateAdapter.isValid(date);
     date = this._getValidDateOrNull(date);
 
-    if (!this._dateAdapter.sameDate(date, this._value)) {
-      this._value = date;
+    const hasChanged = !this._dateAdapter.sameDate(date, this._value);
+
+    // We need to fire the CVA change event for all
+    // nulls, otherwise the validators won't run.
+    if (!date || hasChanged) {
       this._cvaOnChange(date);
-      this._valueChange.emit(date);
       this.dateInput.emit(new MatDatepickerInputEvent(this, this._elementRef.nativeElement));
```

Comment:
@mmalerba I was on the fence on whether this should also be emitted when the value stays null. I ended up doing it because its description implies that it matches the native `input` event. I'm fine with keeping the old behavior where it only fires when something changes, if you think otherwise.

Reply:
Hmm ok. I think we have to make this a major change then and we can see how the presubmit goes. If it's not too difficult to change I agree that it should be changed.